package/MDAnalysis/analysis/encore/utils.py

```diff
@@ -0,0 +1,444 @@
+        Size of the matrix (number of rows or columns). If an array is
+        provided instead, the size of the triangular matrix will be
+        calculated and the array copied as the matrix elements. Otherwise,
+        the matrix is just initialized to zero.
+
+            `metadata` : dict or None
+        Metadata dictionary. Used to generate the metadata attribute.
+
+            `loadfile` : str or None
+        Load the matrix from this file. All the attributes and data will
+        be determined by the matrix file itself (i.e. metadata will be
+        ignored); size has to be provided though.
+        """
+        self.metadata = metadata
+        self.size = size
+        if loadfile:
+            self.loadz(loadfile)
+            return
+        if type(size) == int:
+            self.size = size
+            self._elements = np.zeros((size + 1) * size / 2, dtype=np.float64)
+            return
+        if type(size) == SynchronizedArray:
+            self._elements = np.array(size.get_obj(), dtype=np.float64)
+            self.size = int((np.sqrt(1 + 8 * len(size)) - 1) / 2)
+            return
+        else:
+            raise TypeError
+
+    def __getitem__(self, args):
+        x, y = args
+        if x < y:
+            x, y = y, x
+        return self._elements[x * (x + 1) / 2 + y]
+
+    def __setitem__(self, args, val):
+        x, y = args
+        if x < y:
+            x, y = y, x
+        self._elements[x * (x + 1) / 2 + y] = val
+
+    def as_array(self):
+        """Return standard numpy array equivalent"""
+        a = np.zeros((self.size, self.size))
+        a[np.tril_indices(self.size)] = self._elements
+        a[np.triu_indices(self.size)] = a.T[np.triu_indices(self.size)]
+        return a
+
+    def savez(self, fname):
+        """Save matrix in the npz compressed numpy format. Save metadata and
+        data as well.
+
+        Parameters
+        ----------
+
+        `fname` : str
+            Name of the file to be saved.
+        """
+        np.savez(fname, elements=self._elements, metadata=self.metadata)
+
+    def loadz(self, fname):
+        """Load matrix from the npz compressed numpy format.
+
+        Parameters
+        ----------
+
+        `fname` : str
+            Name of the file to be loaded.
+        """
+        loaded = np.load(fname)
+
+        if loaded['metadata'].shape != ():
+            if loaded['metadata']['number of frames'] != self.size:
+                raise TypeError
+            self.metadata = loaded['metadata']
+        else:
+            if self.size*(self.size-1)/2+self.size != len(loaded['elements']):
+                raise TypeError
+        self._elements = loaded['elements']
+
+    def __mul__(self, scalar):
+        """Multiply with scalar.
+
+        Parameters
+        ----------
+
+        `scalar` : float
+            Scalar to multiply with.
+        """
+        newMatrix = TriangularMatrix(self.size)
```

Comment:
Ah yeah, it does need to return a new object, you're right - I'm thinking of everything as an `__iadd__`.  
If you want to be super safe you could write this as `newMatrix = self.__class__(self.size)` which might save someone a headache if they ever subclass this.